fe_rtos/src/interrupt.rs

```diff
@@ -81,7 +86,33 @@ unsafe extern "C" fn Reset() -> ! {
 pub static RESET_VECTOR: unsafe extern "C" fn() -> ! = Reset;
 
 #[panic_handler]
-fn panic(_panic: &PanicInfo<'_>) -> ! {
+fn panic(panic_info: &PanicInfo<'_>) -> ! {
+    let pid = unsafe { get_cur_task().pid };
+
+    unsafe {
+        super::disable_interrupts();
+    }
+
+    //Output which process has panicked
+    print_msg(format!("Oh no! Process {} has panicked!\n", pid).as_str());
+
+    //Output the file and line number if there is one
+    if let Some(loc) = panic_info.location() {
+        print_msg(format!("{} line {}", loc.file(), loc.line()).as_str());
+    }
+    putc('\n');
```

Comment:
Can we just put these `\n` characters at the end of the format string instead of a separate putc call? In this case, it looks like it prints an extra newline, even if there's no location info, which I'm not sure is intended behavior.